main/src/main/java/org/openrefine/commands/expr/PreviewExpressionCommand.java

```diff
@@ -201,7 +201,12 @@ public void doPost(HttpServletRequest request, HttpServletResponse response)
                 Properties bindings = ExpressionUtils.createBindings();
 
                 if (Mode.RowBased.equals(engine.getMode())) {
-                    List<IndexedRow> rows = state.getRows(engine.combinedRowFilters(), sortingConfig, 0, limit);
+                    GridState sorted = state;
+                    if (!SortingConfig.NO_SORTING.equals(sortingConfig)) {
+                        // TODO cache appropriately
+                        sorted = state.reorderRows(sortingConfig, false);
```

Comment:
This is the sort of place that would benefit from the caching I mentioned above.